data_steward/analytics/cdr_ops/incremental_load_qc.py

```diff
@@ -0,0 +1,348 @@
+    f"{new_dataset}.observation<br><br>")
+failure_null_check = (
+    f"There are <b>{len(df)}</b> records in {new_dataset}.survey_conduct that miss "
+    f"corresponding records {new_dataset}.observation. <br>Look at the table and "
+    "investigate why they are inconsistent.<br><br>")
+
+render_message(df, success_null_check, failure_null_check)
+
+# -
+
+# ## QC 6. Confirm most of "missing basics" issues are remediated
+# This hotfix will fix the "missing basics" problem. But not all "missing basics"
+# will be fixed. <br>This QC is to see how much of the problem is resolved.
+# (We do not have a specific number for this check to succeed/fail.)<br>
+# We must re-assess our hotfix if unresolved "missing basics" are still way too many.
+
+# +
+query = JINJA_ENV.from_string('''
+    WITH new_dataset AS (
+        SELECT COUNT(DISTINCT person_id) AS count_missing_basics
+        FROM `{{project}}.{{new_dataset}}.person`
+        WHERE person_id NOT IN
+        (
+            SELECT DISTINCT person_id
+            FROM `{{project}}.{{new_dataset}}.concept` 
+            JOIN `{{project}}.{{new_dataset}}.concept_ancestor`
+                ON concept_id = ancestor_concept_id
+            JOIN `{{project}}.{{new_dataset}}.observation`
+                ON descendant_concept_id = observation_concept_id
+            WHERE observation_concept_id NOT IN (40766240, 43528428, 1585389)
+            AND concept_class_id = 'Module'
+            AND concept_name IN ('The Basics') 
+            AND questionnaire_response_id is not null
+            AND observation_source_value NOT LIKE 'Second%' 
+            AND observation_source_value NOT LIKE 'PersonOne%'
+            AND observation_source_value NOT LIKE 'SocialSecurity%'
+        )
+    ), source_dataset AS (
+        SELECT COUNT(DISTINCT person_id) AS count_missing_basics
+        FROM `{{project}}.{{source_dataset}}.person`
+        WHERE person_id NOT IN
+        (
+            SELECT DISTINCT person_id
+            FROM `{{project}}.{{source_dataset}}.concept` 
+            JOIN `{{project}}.{{source_dataset}}.concept_ancestor`
+                ON concept_id = ancestor_concept_id
+            JOIN `{{project}}.{{source_dataset}}.observation`
+                ON descendant_concept_id = observation_concept_id
+            WHERE observation_concept_id NOT IN (40766240, 43528428, 1585389)
+            AND concept_class_id = 'Module'
+            AND concept_name IN ('The Basics') 
+            AND questionnaire_response_id is not null
+            AND observation_source_value NOT LIKE 'Second%' 
+            AND observation_source_value NOT LIKE 'PersonOne%'
+            AND observation_source_value NOT LIKE 'SocialSecurity%'
+        )
+    )
+    SELECT 
+        new_dataset.count_missing_basics AS remaining_missing_basics,
+        source_dataset.count_missing_basics AS original_missing_basics,
+        source_dataset.count_missing_basics - new_dataset.count_missing_basics AS resolved_missing_basics
```

Comment:
@hiro-mishima I like most of this check.  Would it be too much to ask to include the percentage of total participants with only junk basics data?

Reply:
I included the percentage with this commit: https://github.com/all-of-us/curation/pull/1489/commits/bcfb868c87771b9794b8dd44f7956d0301960c56